solarwinds_apm/oboe/oboe_sampler.py

```diff
@@ -0,0 +1,702 @@
+        sample_state.trace_options = TraceOptionsWithResponse(
+            trigger_trace=parsed.trigger_trace,
+            timestamp=parsed.timestamp,
+            sw_keys=parsed.sw_keys,
+            custom=parsed.custom,
+            ignored=parsed.ignored,
+            response=TraceOptionsResponse(
+                auth=None, trigger_trace=None, ignored=None
+            ),
+        )
+        self.logger.debug(
+            "X-Trace-Options present %s", sample_state.trace_options
+        )
+        if sample_state.headers.x_trace_options_signature:
+            sample_state.trace_options.response.auth = validate_signature(
+                sample_state.headers.x_trace_options,
+                sample_state.headers.x_trace_options_signature,
+                (
+                    sample_state.settings.signature_key
+                    if sample_state.settings
+                    and sample_state.settings.signature_key
+                    else None
+                ),
+                sample_state.trace_options.timestamp,
+            )
+            if sample_state.trace_options.response.auth != Auth.OK:
+                self.logger.debug(
+                    "X-Trace-Options-Signature invalid; tracing disabled"
+                )
+                new_trace_state = self.set_response_headers_from_sample_state(
+                    sample_state,
+                    parent_context,
+                    trace_id,
+                    name,
+                    kind,
+                    attributes,
+                    links,
+                    trace_state,
+                )
+                return SamplingResult(
+                    decision=Decision.DROP, trace_state=new_trace_state
+                )
+        if not sample_state.trace_options.trigger_trace:
+            sample_state.trace_options.response.trigger_trace = (
+                TriggerTrace.NOT_REQUESTED
+            )
+        if sample_state.trace_options.sw_keys:
+            sample_state.attributes[SW_KEYS_ATTRIBUTE] = (
+                sample_state.trace_options.sw_keys
+            )
+        for key, value in sample_state.trace_options.custom.items():
+            sample_state.attributes[key] = value
+        if len(sample_state.trace_options.ignored) > 0:
+            sample_state.trace_options.response.ignored = [
+                key for key, _ in sample_state.trace_options.ignored
+            ]
+        return None
+
+    def _handle_no_settings(
+        self,
+        sample_state: SampleState,
+        parent_context: "Context" | None,
+        trace_id: int,
+        name: str,
+        kind: SpanKind | None,
+        attributes: Attributes,
+        links: Sequence["Link"] | None,
+        trace_state: "TraceState" | None,
+    ) -> "SamplingResult":
+        """
+        Handle the case where settings are unavailable.
+        """
+        self.logger.debug("settings unavailable; sampling disabled")
```

Comment:
I'd say this one is important enough to log as warning:

```suggestion
        self.logger.warning("settings unavailable; sampling disabled")
```